Source/SwiftLintFramework/Rules/Idiomatic/ConvenienceTypeRule.swift

```diff
@@ -87,11 +131,35 @@ public struct ConvenienceTypeRule: ASTRule, OptInRule, ConfigurationProviderRule
             return []
         }
 
-        return [
-            StyleViolation(ruleDescription: type(of: self).description,
-                           severity: configuration.severity,
-                           location: Location(file: file, byteOffset: offset))
-        ]
+        // MARK: Has @objc members?
```

Comment:
doesn't the class need to explicitly inherit from `NSObject` (or a subclass of that) in order for this to be useful? if so, that should already be covered by the "Inherits from anything?" check

Reply:
It doesn't appear to. This works correctly in a playground:

```swift
import UIKit

final class MyClass {
    @objc class func foo(sender: UIButton) {
        print("hello")
    }
}

let button = UIButton()
button.setTitle("Say Hello", for: .normal)
button.addTarget(MyClass.self, action: #selector(MyClass.foo(sender:)), for: .touchUpInside)
button.sizeToFit()

import PlaygroundSupport
PlaygroundPage.current.liveView = button
```